ydb/docs/en/core/integrations/grafana.md

```diff
@@ -0,0 +1,136 @@
+
+### Authentication
+
+The Grafana plugin supports various authentication [authentication methods](../reference/ydb-sdk/auth.md).
+
+Below is an example config for authenticating a {{ ydb-short-name }} data source using username and password:
+
+```yaml
+apiVersion: 1
+datasources:
+  - name: YDB
+    type: ydbtech-ydb-datasource
+    jsonData:
+      authKind: "UserPassword"
+      endpoint: 'grpcs://<hostname>:2135'
+      dbLocation: '/eu-central1/b1g/db-location'
+      user: '<username>'
+    secureJsonData:
+      password: '<userpassword>'
+      certificate: 'content of *.pem file'
+```
+
+Here are fields that are supported in connection configuration:
+
+| Name  | Description         |         Type          |
+| :---- | :------------------ | :-------------------: |
+| authKind | Authentication type |       `"Anonymous" \| "ServiceAccountKey" \| "AccessToken" \| "UserPassword" \| "MetaData"`        |
+| endpoint | Database endpoint  | `string` |
+| dbLocation | Database location  | `string` |
+| user | User name  | `string` |
+| serviceAccAuthAccessKey | Service account access key  | `string` (secured) |
+| accessToken | OAuth access token  | `string` (secured) |
+| password | User password  | `string` (secured) |
+| certificate | Content of *.pem file  | `string` (secured) |
+
+## Building queries
+
+{{ ydb-short-name }} is queried with a SQL dialect named [YQL](../yql/reference/index.md).
+Queries can contain macros which simplify syntax and allow for dynamic parts. It may be two kind of macro - [grafana-level](#macros) and ydb-level. Plugin will parse query text and, before sending it to YDB, substitute variables and grafana-level macroses with particular values. After that ydb-level macroses will be treated by ydb. 
+The query editor allows you to get data in different representation: time series, table or logs.
+
+### Time series (../_assets/grafana/time-series.png)
+
+Time series visualization options are selectable if the query returns at least one field with `Date`, `Datetime`, or `Timestamp` type type and at least one field with `int`, `uint`, `double` or `float` type. Then you can select time series visualization options. Grafana interprets timestamp rows without explicit time zone as UTC. Any other column is treated as a value column.
+
+#### Multi-line time series
+
+To create multi-line time series, the query must return at least 3 fields in the following order:
+
+- field 1: `Date`, `Datetime` or `Timestamp`
+- field 2: value to group by
+- field 3+: the metric values
+
+For example:
+
+```sql
+SELECT `timestamp`, `requestTime`, AVG(`responseStatus`) AS `avgRespStatus`
+FROM `/database/endpoint/my-logs`
+GROUP BY `requestTime`, `timestamp`
+ORDER BY `timestamp`
+```
+
+### Tables (../_assets/grafana/table.png)
+
+Table visualizations will always be available for any valid {{ ydb-short-name }} query (multi result sets are not supported).
```

Comment:
what will happen for queries that do not return a result set like `INSERT INTO`?

Reply:
Modifying queries are not supported from within Grafana.